File: rails/app/models/event.rb
class Event < ActiveRecord::Base
  extend FriendlyId
  friendly_id :slug_candidates, use: :slugged

  def slug_candidates
    [
      [:name]
    ]
  end

  belongs_to :comp
  belongs_to :discipline
  belongs_to :gender
  has_many :rounds, through: :event_rounds
  has_many :event_rounds, dependent: :destroy
  has_many :registrations, dependent: :destroy

  validates_presence_of :comp, :discipline, :gender, :name
end

File: ember/app/models/event.js
import DS from 'ember-data';
import Ember from 'ember';

export default DS.Model.extend(Ember.Validations.Mixin, {
  name: DS.attr('string'),
  slug: DS.attr('string'),

  comp: DS.belongsTo('comp', { async: true }),
  discipline: DS.belongsTo('discipline', { async: true }),
  rounds: DS.hasMany('round', { async: true }),
  registrations: DS.hasMany('registration', { async: true }),
  gender: DS.belongsTo('gender', { async: true })//,

  // validations: {
  //     name: {
  //       presence: true
  //     },
  //     discipline: {
  //       presence: true
  //     },
  //     gender: {
  //       presence: true
  //     }
  //   }
});
